src/components/application_manager/include/application_manager/rpc_passing_handler.h

```diff
@@ -0,0 +1,106 @@
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_RPC_PASSING_HANDLER_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_RPC_PASSING_HANDLER_H_
+
+#include "application_manager/application.h"
+#include "interfaces/MOBILE_API.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/app_service_manager.h"
+#include <deque>
+#include "utils/timer.h"
+
+namespace application_manager {
+
+struct ServiceInfo {
+  std::string service_id;
+  uint32_t connection_key;
+};
+
+typedef std::shared_ptr<timer::Timer> TimerSPtr;
+
+class AppServiceManager;
+
+class RPCPassingHandler {
+ public:
+  RPCPassingHandler(AppServiceManager& asm_ref, ApplicationManager& am_ref);
+
+  /**
+   * @brief Check if function id is in the handled_rpcs list of an active app
+   * service
+   * @param function_id RPC function id
+   * @return true if function id exists in handled_rpcs list of an active app
+   * service
+   */
+  bool CanHandleFunctionID(int32_t function_id);
+
+  /**
+   * @brief Check if app services or core is being used to handle the RPC
+   * @param correlation_id correlation id of RPC response
+   * @return true if an app service was used to handle the RPC
+   */
+  bool IsPassThroughMessage(uint32_t correlation_id,
+                            commands::Command::CommandSource source,
+                            int32_t message_type);
+
+  /**
+   * @brief Function to handle sending and receiving RPC Passing
+   * requests/responses
+   * @param rpc_message RPC message SmartObject
+   * @return true to stop current request/response and forward it
```

Comment:
`true if the request was forwarded, false otherwise`

Reply:
Resolved in https://github.com/smartdevicelink/sdl_core/pull/2821/commits/d255c6004c831f4b208c1cec086a7088dafa7196